reports.go

```diff
@@ -46,15 +70,711 @@ type Report struct {
+
+	if countryCriminalSearchID == "" {
+		return nil, errEmptyID
+	}
+
+	rel, err := url.Parse(countryCriminalSearchPath)
+	if err != nil {
+		return nil, err
+	}
+
+	u := *c.BaseURL
+	u.Path = path.Join(u.Path, rel.String(), countryCriminalSearchID)
+
+	req, err := http.NewRequest(http.MethodGet, u.String(), nil)
+	if err != nil {
+		return nil, err
+	}
+	req.Header.Add("Content-Type", "application/json")
+	req.SetBasicAuth(c.APIKey, "")
+
+	resp, err := c.client.Do(req)
+	if err != nil {
+		return nil, err
+	}
+
+	if resp.StatusCode != http.StatusOK {
+		return nil, NewResponseError([]int{
+			resp.StatusCode,
+		}, resp)
+	}
+	defer func() {
+		if resp.Body != nil {
+			resp.Body.Close()
+		}
+	}()
+
+	b, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return nil, err
+	}
+
+	getResp := &CountryCriminalSearch{}
+	err = json.Unmarshal(b, getResp)
+	if err != nil {
+		return nil, err
+	}
+
+	return getResp, nil
+}
+
+const stateCriminalSearchPath = "/v1/state_criminal_searches"
+
+// GetStateCriminalSearch ...
+func (r *Report) GetStateCriminalSearch(stateCriminalSearchID string, c *Client) (*StateCriminalSearch, error) {
+
+	if stateCriminalSearchID == "" {
+		return nil, errEmptyID
+	}
+
+	rel, err := url.Parse(stateCriminalSearchPath)
+	if err != nil {
+		return nil, err
+	}
+
+	u := *c.BaseURL
+	u.Path = path.Join(u.Path, rel.String(), stateCriminalSearchID)
+
+	req, err := http.NewRequest(http.MethodGet, u.String(), nil)
+	if err != nil {
+		return nil, err
+	}
+	req.Header.Add("Content-Type", "application/json")
+	req.SetBasicAuth(c.APIKey, "")
+
+	resp, err := c.client.Do(req)
+	if err != nil {
+		return nil, err
+	}
+
+	if resp.StatusCode != http.StatusOK {
+		return nil, NewResponseError([]int{
+			resp.StatusCode,
+		}, resp)
+	}
+	defer func() {
+		if resp.Body != nil {
+			resp.Body.Close()
+		}
+	}()
+
+	b, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return nil, err
+	}
+
+	getResp := &StateCriminalSearch{}
+	err = json.Unmarshal(b, getResp)
+	if err != nil {
+		return nil, err
+	}
+
+	return getResp, nil
+}
+
+const motorVehicleReportSearchPath = "/v1/motor_vehicle_reports"
+
+// GetMotorVehicleReportSearch ...
+func (r *Report) GetMotorVehicleReportSearch(motorVehicleReportID string, c *Client) (*MotorVehicleReport, error) {
+
+	if motorVehicleReportID == "" {
+		return nil, errEmptyID
+	}
+
+	rel, err := url.Parse(motorVehicleReportSearchPath)
+	if err != nil {
+		return nil, err
+	}
+
+	u := *c.BaseURL
+	u.Path = path.Join(u.Path, rel.String(), motorVehicleReportID)
+
+	req, err := http.NewRequest(http.MethodGet, u.String(), nil)
+	if err != nil {
+		return nil, err
+	}
+	req.Header.Add("Content-Type", "application/json")
+	req.SetBasicAuth(c.APIKey, "")
+
+	resp, err := c.client.Do(req)
+	if err != nil {
+		return nil, err
+	}
+
+	if resp.StatusCode != http.StatusOK {
+		return nil, NewResponseError([]int{
+			resp.StatusCode,
```

Comment:
@chandradeepak the status here should be the `expected` status code, so it should be `http.StatusOK`.